frontend/src/components/TreeViewer/TreeViewer.jsx

```diff
@@ -261,19 +262,22 @@ function getParent(nodeName, nameToNodeMapping) {
  * being viewed in the infoViewer; "" if nothing currently viewed
  * @param {bool} stayOpen: Whether to stay open if the infoViewer is already open;
  * defaults to false
+ * @param {bool} closeButtonClicked: Whether the current signal comes from the close
```

Comment:
I think we can merge `stayOpen` and `closeButtonClicked` into one parameter, like a "force" parameter that can either be `FORCE_OPEN`, `FORCE_CLOSE`, or `FORCE_NONE` which can help us define what this function does in different situations, because right now there is a conflict if both `closeButtonCLicked` and `stayOpen` are true. Alternatively we can factor out the actual logic in this function to a `closeInfobox` and `openInfobox` function and also keep this `toggleInfoBox` which calls `closeInfoBox` and `openInfoBox` in different situations. I like the 2nd option better; what do you think?

Reply:
Need more clarification on this one ples, how are their functions comparable to each other?

Reply:
Update: I and @sultansidhu got on call yesterday and refactored this to make it much better